src/main/java/org/reaktivity/nukleus/tcp/internal/reader/stream/StreamFactory.java

```diff
@@ -114,7 +106,16 @@ private int handleRead() throws IOException
             readBuffer.position(0);
             readBuffer.limit(limit);
 
-            int bytesRead = channel.read(readBuffer);
+            int bytesRead;
+            try
+            {
+                bytesRead = channel.read(readBuffer);
+            }
+            catch(IOException ex)
+            {
+                // RST from client. We treat this as end of stream (see issue #9).
```

Comment:
Please avoid referencing specific issues from the code, just document the intent instead (if not already clear from the code).
ie. `// treat TCP reset as end-of-stream`